src/Pdp/PublicSuffixListManager.php

```diff
@@ -170,25 +229,45 @@ public function buildArray(array &$publicSuffixListArray, array $ruleParts)
      */
     public function writePhpCache(array $publicSuffixList)
     {
-        $data = '<?php' . PHP_EOL . 'return ' . var_export($publicSuffixList, true) . ';';
+        return $this->varExportToFile(self::PDP_PSL_PHP_FILE, $publicSuffixList);
+    }
 
-        return $this->write(self::PDP_PSL_PHP_FILE, $data);
+    /**
+     * Writes php array representation to disk.
+     *
+     * @param string $basename file path
+     * @param array  $input    input data
+     *
+     * @return int Number of bytes that were written to the file
+     */
+    protected function varExportToFile($basename, array $input)
+    {
+        $data = '<?php' . PHP_EOL . 'return ' . var_export($input, true) . ';';
+
+        return $this->write($basename, $data);
     }
 
     /**
      * Gets Public Suffix List.
      *
+     * @param string|null $section the Public Suffix List type
+     *
      * @return PublicSuffixList Instance of Public Suffix List
      */
-    public function getList()
+    public function getList($section = null)
```

Comment:
After adding a constant for `ALL` (or `ALL_DOMAINS`), the new constant should replace `null` as the default argument.